### PR TITLE
Implemented code for article

### PR DESCRIPTION
**Pull request by sagaofsilence**
Added model, controller, exception handler and tests

### CODE REVIEW

File: spring-boot-modules/spring-boot-mvc-5/src/main/java/com/baeldung/problemdetails/web/SalesController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("sales")
+public class SalesController {
+
+  @PostMapping("/calculate")
+  public ResponseEntity<OperationResult> calculate(
+      @Validated @RequestBody OperationRequest operationRequest) {
+    OperationResult operationResult = null;
```
Comment (JavierPeris) [inline spring-boot-modules/spring-boot-mvc-5/src/main/java/com/baeldung/problemdetails/web/SalesController.java:20]: No need to initialize to null the variable.

**Review by JavierPeris — CHANGES_REQUESTED** [review]
Please apply Baeldung formatter.

File: spring-boot-modules/spring-boot-mvc-5/src/main/java/com/baeldung/problemdetails/web/SalesController.java (hunk 1 above)
Comment (sagaofsilence) [reply spring-boot-modules/spring-boot-mvc-5/src/main/java/com/baeldung/problemdetails/web/SalesController.java]: Done

**Comment by sagaofsilence** [comment]
image
I am using formatter

**Comment by sagaofsilence** [comment]
Fixed the code. Also fixed formatting using baeldung formatter. Please check. @JavierPeris

File: spring-boot-modules/spring-boot-mvc-5/src/main/java/com/baeldung/problemdetails/web/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.baeldung.problemdetails.web;
+
+import com.baeldung.problemdetails.model.InvalidInputException;
+import java.net.URI;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
```
Comment (JavierPeris) [inline spring-boot-modules/spring-boot-mvc-5/src/main/java/com/baeldung/problemdetails/web/GlobalExceptionHandler.java:10]: An example that the formatter is not being applied is the order of the imports and the separation between the different groups:
Java

Jakarta

javax

organization

com

Comment (sagaofsilence) [reply spring-boot-modules/spring-boot-mvc-5/src/main/java/com/baeldung/problemdetails/web/GlobalExceptionHandler.java]: Not sure why in Intellij code formatter is not organizing the imports. I used Eclipse to format the imports for now.

**Pull request merged**